### DISCUSSION TITLE
Profile Update failing for  freshly setup Sunbird Instance

### DISCUSSION THREAD

**Discussion by kiranharidas187** [Sunbird-Ed/Community#3] [Installation] [answered]
@sharathkashyap @surendrasinghs 

We are setting up a new flavour of SunbirdEd`(release-4.6.0)` for the Shikshalokam organization. As part of this activity, we are stuck at one point. 

**Issue**

As an administrator, we are not able to update the profile. As per the flow, the admin should be able to select a state, Sub role, district, block, cluster, etc till the lowest entity in the hierarchy. We have made all the necessary changes to the form to render the profile update page and the user is able to select the fields. But on calling the profile update it is returning the following error .
`Invalid value hm for parameter userSubType. Please provide a valid value.`

**Solution Tried**
We tried updating the same sunbird staging form config(profileConfig_v2) in the new environment but still, the same issue exists.

Please let us know how it can be fixed. Attaching complete curl and response for ref

```
curl '<BASE_URL>/learner/user/v3/update' \
  -X 'PATCH' \
  -H 'Accept: application/json' \
  -H 'Accept-Language: en-US,en;q=0.9' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: application/json' \
  -H 'Cookie: fpestid=ffZoqj5bAokAqB_3oAGZXt3qUBGDVzbZXZw0vf_djJAwB_IhvOVwt-VB4v_lno_R48UBxw; connect.sid=s%3Ac-Z0TSm-GxgcL1VlRjGM0xKv49QR1uLg.tdqDDcNpX23IvDuhFpyTZ%2B6juh4xlA05Wx3LbR0z9qg' \
  -H 'Origin: https://dhruva.shikshalokam.org' \
  -H 'Referer: https://dhruva.shikshalokam.org/profile' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.88 Safari/537.36' \
  -H 'X-App-Id: slecosystem.slecosystem.portal' \
  -H 'X-App-Version: 4.6.0' \
  -H 'X-Channel-Id: 0124487522476933120' \
  -H 'X-Device-ID: f443fa7035c02296bcb44ba15c109950' \
  -H 'X-Org-code: 0124487522476933120' \
  -H 'X-Request-ID: 10da51f4-1e44-a235-5401-d725ec0aa683' \
  -H 'X-Session-ID: c-Z0TSm-GxgcL1VlRjGM0xKv49QR1uLg' \
  -H 'X-Source: web' \
  -H 'X-User-ID: 0edbd37c-5613-47c5-90cd-f21898a08b25' \
  -H 'X-msgid: 10da51f4-1e44-a235-5401-d725ec0aa683' \
  -H 'sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="100", "Google Chrome";v="100"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'ts: 2022-04-13T13:54:45+05:30' \
  --data-raw '{"params":{},"request":{"userId":"0edbd37c-5613-47c5-90cd-f21898a08b25","profileLocation":[{"code":"KA","name":"Karnataka","id":"adf45b4d-9048-4e96-b04f-1c4845bd6488","type":"state"},{"code":"BLR-U","name":"Bangalore Urban","id":"d0793ea2-7aa8-4b2b-abd7-16e06c80ebfc","type":"district","parentId":"adf45b4d-9048-4e96-b04f-1c4845bd6488"}],"profileUserTypes":[{"type":"administrator","subType":"hm"}],"firstName":"Support"}}' \
  --compressed
```

Response

```
{
  "id": "api.user.update",
  "ver": "v3",
  "ts": "2022-04-13 08:24:45:627+0000",
  "params": {
    "resmsgid": null,
    "msgid": "10da51f4-1e44-a235-5401-d725ec0aa683",
    "err": "INVALID_PARAMETER_VALUE",
    "status": "INVALID_PARAMETER_VALUE",
    "errmsg": "Invalid value hm for parameter userSubType. Please provide a valid value."
  },
  "responseCode": "CLIENT_ERROR",
  "result": {}
}
```

cc: @raghupathiguduri @vijayashree@shikshalokam.org

**Comment by kiranharidas187** [comment, marked as answer]
Can anyone help with this?
cc- @Krishnaj20 @vijiurs

**Reply by Krishnaj20** [reply]
Tagging @luckynvdu & @sharathkashyap from SunbirdEd BB community.

**Reply by luckynvdu** [reply]
@rhwarrier @reshmi-nair Can you please respond to this as it looks like a Sunbird lern issue.

**Reply by reshmi-nair** [reply]
@kiranharidas187 Please try updating the 'profileConfig' instead of 'profileConfig_v2'. 'profileConfig_v2' is used from release-4.7.0. Also after updating 'profileConfig' restart the learner pod before testing.

**Reply by kiranharidas187** [reply]
@reshmi-nair This solution worked.  Thank you :)
